py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -240,17 +250,18 @@ def write_and_archive(self, table: pyarrow.table) -> None:
             self.error_files += self.archive_files
             self.archive_files = []
 
-        finally:
-            if len(self.error_files) > 0:
-                move_s3_objects(
-                    self.error_files,
-                    os.path.join(os.environ["ERROR_BUCKET"], DEFAULT_S3_PREFIX),
-                )
+    def move_s3_files(self) -> None:
+        """
+        move archive and error files to their respective s3 buckets.
+        """
+        if len(self.error_files) > 0:
+            move_s3_objects(
+                self.error_files,
+                os.path.join(os.environ["ERROR_BUCKET"], DEFAULT_S3_PREFIX),
+            )
```

Comment:
break out move into it's own function, this will probably have to develop more to isolate possible failure points during ingestion process